manifests/instance.pp

```diff
@@ -501,118 +501,139 @@
+    set_max_intset_entries        => $set_max_intset_entries,
+    zset_max_ziplist_entries      => $zset_max_ziplist_entries,
+    zset_max_ziplist_value        => $zset_max_ziplist_value,
+    hll_sparse_max_bytes          => $hll_sparse_max_bytes,
+    activerehashing               => $activerehashing,
+    output_buffer_limit_slave     => $output_buffer_limit_slave,
+    output_buffer_limit_pubsub    => $output_buffer_limit_pubsub,
+    hz                            => $hz,
+    aof_rewrite_incremental_fsync => $aof_rewrite_incremental_fsync,
+    cluster_enabled               => $cluster_enabled,
+    cluster_config_file           => $cluster_config_file,
+    cluster_node_timeout          => $cluster_node_timeout,
+    cluster_slave_validity_factor => $cluster_slave_validity_factor,
+    cluster_require_full_coverage => $cluster_require_full_coverage,
+    cluster_migration_barrier     => $cluster_migration_barrier,
+    extra_config_file             => $extra_config_file,
+    tls_port                      => $tls_port,
+    tls_cert_file                 => $tls_cert_file,
+    tls_key_file                  => $tls_key_file,
+    tls_ca_cert_file              => $tls_ca_cert_file,
+    tls_ca_cert_dir               => $tls_ca_cert_dir,
+    tls_ciphers                   => $tls_ciphers,
+    tls_ciphersuites              => $tls_ciphersuites,
+    tls_protocols                 => $tls_protocols,
+    tls_auth_clients              => $tls_auth_clients,
+    tls_replication               => $tls_replication,
+    tls_cluster                   => $tls_cluster,
+    tls_prefer_server_ciphers     => $tls_prefer_server_ciphers,
+    modules                       => $modules,
+    io_threads                    => $io_threads,
+    io_threads_do_reads           => $io_threads_do_reads,
+    cluster_allow_reads_when_down => $cluster_allow_reads_when_down,
+    cluster_replica_no_failover   => $cluster_replica_no_failover,
+    dynamic_hz                    => $dynamic_hz,
+    activedefrag                  => $activedefrag,
+    active_defrag_ignore_bytes    => $active_defrag_ignore_bytes,
+    active_defrag_threshold_lower => $active_defrag_threshold_lower,
+    active_defrag_threshold_upper => $active_defrag_threshold_upper,
+    active_defrag_cycle_min       => $active_defrag_cycle_min,
+    active_defrag_cycle_max       => $active_defrag_cycle_max,
+    active_defrag_max_scan_fields => $active_defrag_max_scan_fields,
+    jemalloc_bg_thread            => $jemalloc_bg_thread,
+    rdb_save_incremental_fsync    => $rdb_save_incremental_fsync,
+    acls                          => $acls,
+    custom_options                => $custom_options,
+  }
+
+  # Rely on https://github.com/puppetlabs/puppetlabs-stdlib/pull/1425
+  # once available.
+  if $_template_params.stdlib::nested_values.any |$value| { $value.is_a(Deferred) } {
+    $_template_params_escaped = $_template_params.map | $_var , $_value | {
+      if $_value.is_a(Deferred) {
```

Comment:
This is a bit inconsistent. `nested_values` does a deep collection, also going into hashes while the inner part only does the values of the map.